src/main/groovy/com/github/lookout/whoas/RedisQueue.groovy

```
@@ -0,0 +1,138 @@
+package com.github.lookout.whoas
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import redis.clients.jedis.Jedis;
+
+/**
+ * A redis queue that offers distributed and persistent queue
+ */
+class RedisQueue extends AbstractHookQueue {
+    private final RedisClientFactory redisClientFactory
+    private static nextId = 0
+    private String hostname
+    private Integer port
+
+    /**
+     * Create the RedisQueue with valid config
+     */
+    RedisQueue(WhoasQueueConfig queueConfig) {
+        redisClientFactory = RedisClientFactory.getInstance()
+        this.hostname = queueConfig.hostname
+        this.port = queueConfig.port
+    }
+
+    /**
+     * Default constructor
+     */
+    RedisQueue() {
+        redisClientFactory = RedisClientFactory.getInstance()
+        this.hostname = "localhost"
+        this.port = 6379
+    }
+    /**
+     * Return the number of elements in the queue
+     */
+    Long getSize() {
+        if (!this.started) {
+            throw new Exception("Queue must be started before this operation is invoked")
+        }
+        Jedis redisclient = null
+        Long queueSize = 0
```

We are handling the exception in this function. So we are returning "0" in case of errors. Instead of returning -1, we should throw exception, but thats not the intent here